components/Cards/src/MDCCollectionViewCardCell.m

```diff
@@ -0,0 +1,209 @@
+
+@interface MDCCollectionViewCardCell ()
+
+@property(nonatomic, strong, nullable) UIImageView *selectedImageView;
+@property(nonatomic, assign) CGPoint lastTouch;
+
+@end
+
+@implementation MDCCollectionViewCardCell {
+  BOOL _inkAnimating;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame {
+  self = [super initWithFrame:frame];
+  if (self) {
+    [self commonMDCCollectionViewCardCellInit];
+  }
+  return self;
+}
+
+- (void)commonMDCCollectionViewCardCellInit {
+  _cardView = [[MDCCardView alloc] initWithFrame:self.contentView.bounds];
+  _cardView.autoresizingMask =
+    UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+  _cardView.userInteractionEnabled = NO;
+  self.contentView.autoresizingMask =
+    UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+  [self.contentView addSubview:self.cardView];
+  _inkAnimating = NO;
+  self.selectionMode = NO;
+
+  [self initializeSelectedImage];
+
+  self.cornerRadius = 4.f;
+}
+
+- (void)initializeSelectedImage {
+  UIImage *circledCheck = [MDCIcons imageFor_ic_check_circle];
+  circledCheck = [circledCheck imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate];
+  self.selectedImageView = [[UIImageView alloc] initWithImage:circledCheck];
+  self.selectedImageView.center = CGPointMake(
+                                    CGRectGetWidth(self.bounds) - (circledCheck.size.width/2) - 8,
+                                    (circledCheck.size.height/2) + 8);
+  self.selectedImageView.layer.zPosition = MAXFLOAT - 1;
+  self.selectedImageView.autoresizingMask =
+  (UIViewAutoresizingFlexibleRightMargin | UIViewAutoresizingFlexibleTopMargin |
+   UIViewAutoresizingFlexibleLeftMargin | UIViewAutoresizingFlexibleBottomMargin);
+  [self.contentView addSubview:self.selectedImageView];
+  self.selectedImageView.hidden = YES;
+}
+
+- (void)setBackgroundColor:(UIColor *)backgroundColor {
+  super.backgroundColor = backgroundColor;
+  self.cardView.backgroundColor = backgroundColor;
+
+  /**
+   currently the selected check image uses the color
+   based on MDFTextAccessibility to fit the background color.
+   */
+  UIColor *checkColor =
+  [MDFTextAccessibility textColorOnBackgroundColor:backgroundColor
+                                   targetTextAlpha:1.f
+                                           options:MDFTextAccessibilityOptionsNone];
+  self.selectedImageTintColor = checkColor;
+}
+
+- (UIColor *)backgroundColor {
+  return self.cardView.backgroundColor;
+}
+
+- (void)setCornerRadius:(CGFloat)cornerRadius {
+  self.layer.cornerRadius = cornerRadius;
+  self.cardView.cornerRadius = cornerRadius;
+  [self setNeedsLayout];
+}
+
+- (CGFloat)cornerRadius {
+  return self.cardView.cornerRadius;
+}
+
+- (void)selectionState:(MDCCardCellSelectionState)state withAnimation:(BOOL)animation {
+  _selectionState = state;
+  self.selectionMode = YES;
+  switch (state) {
+    case MDCCardCellSelectionStateSelected: {
+      if (animation) {
+        _inkAnimating = YES;
```

Comment:
When does this get set back to `NO` when the animation finishes?

Reply:
improved